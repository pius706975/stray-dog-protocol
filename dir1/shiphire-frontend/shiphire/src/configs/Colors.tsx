export const Color = {
    bgColor: "#f0f6fb",
    softSecBgPrimary: "#bcd3e6",
    softGreyBgPrimary: "#cbcdd1",
    primaryColor: "#216dab",
    softPrimaryColor: "#6499c4",
    secColor: "#9cd2ff",
    softSecColor: "#d7edff",
    darkTextColor: "#4d5963",
    lightTextColor: "#f0f6fb",
    bgSuccessColor: "#e9faec",
    boldSuccessColor: "#1fcc44",
    successColor: "#4cd669",
    bgErrorColor: "#fae9e7",
    boldErrorColor: "#cc1f12",
    errorColor: "#d64c41",
    bgInfoColor: "#d0dbf5",
    boldInfoColor: "#124ccc",
    infoColor: "#5982db",
    bgWarningColor: "#fbf6e8",
    boldWarningColor: "#d9a318",
    warningColor: "#e1b546",
    bgNeutralColor: "#dbdee0",
    neutralColor: "#828b92",
    primaryDisableColor: "#6499c4"
}