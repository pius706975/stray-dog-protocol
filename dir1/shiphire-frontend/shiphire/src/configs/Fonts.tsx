export const FontFamily = {
    bold: "Kanit-Bold",
    boldItalic: "Kanit-BoldItalic",
    medium: "Kanit-Medium",
    mediumItalic: "Kanit-MediumItalic",
    regular: "Kanit-Regular",
    semiBold: "Kanit-SemiBold",
    semiBoldItalic: "Kanit-SemiBoldItalic"
}

export const FontSize = {
    xs: 14,
    sm: 16,
    md: 18,
    lg: 20,
    xl: 22,
    xl1: 24,
    xl2: 26,
    xxl: 30,
    xxxl: 36
}