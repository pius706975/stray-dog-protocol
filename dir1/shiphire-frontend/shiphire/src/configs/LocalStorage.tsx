import AsyncStorage from '@react-native-async-storage/async-storage';

export const TOKEN = '@token';
export const ROLES = '@roles';
export const ADMINDATA = '@admindata';
export const USERDATA = '@userdata';
export const COMPANYDATA = '@companydata';
export const SHIPOWNERDOCSUBMITTED = '@shipownerdocsubmitted';
export const RFQFILEPATH = '@rfqfilepath';
export const PROPOSALOWNERFILEPATH = '@proposalfilepath';
export const CONTRACTFILEPATH = '@contractfilepath';
export const USERLANGUAGE = '@userlanguage';
export const FCMTOKEN = '@fcmtoken';

export const setDataToLocalStorage = async (key: string, data: Object) => {
    try {
        const jsonData = JSON.stringify(data);
        return await AsyncStorage.setItem(key, jsonData);
    } catch (error) {
        return error;
    }
};

export const getDataFromLocalStorage = async (key: string) => {
    try {
        const jsonValue = await AsyncStorage.getItem(key);
        let valueObj;
        if (jsonValue) valueObj = JSON.parse(jsonValue);
        return valueObj;
    } catch (error) {
        return error;
    }
};

export const removeDataToLocalStorage = async (key: string) => {
    try {
        return await AsyncStorage.removeItem(key);
    } catch (error) {
        return error;
    }
};
