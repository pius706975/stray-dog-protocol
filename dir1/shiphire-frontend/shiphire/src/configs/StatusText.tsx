export const StatusText = {
    'rfq 1': 'Waiting for ship owner to respond',
    'rfq 2': 'Waiting on your proposal',
    'proposal 1': 'Waiting for your agreement',
    'proposal 2': 'Waiting for your payment',
    'proposal 3': 'Waiting for payment approval',
    'proposal 4': 'Waiting for contract',
    'contract 1': 'Waiting for your signature',
};
