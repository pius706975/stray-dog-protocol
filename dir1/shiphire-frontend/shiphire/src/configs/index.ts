export * from "./Fonts"
export * from "./Colors"
export * from "./LocalStorage"
export * from "./Svg"
export * from "./StatusText"
export * from "./firebase/GoogleSignIn"
export * from "./firebase/AppleSignIn"