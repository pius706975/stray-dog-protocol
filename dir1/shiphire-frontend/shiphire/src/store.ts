import { configureStore } from '@reduxjs/toolkit';
import {
    addShipSlice,
    documentNameSlice,
    modalSlice,
    networkErrorSlice,
    notifBadgeSlice,
    progressIndicatorSlice,
    renterPreferenceSlice,
    requestForQuoteSlice,
    shipSpecSlice,
    termsStatusSlice,
    userLanguageSlice,
    userLocationSlice,
    userStatusSlice,
} from './slices';

const store = configureStore({
    reducer: {
        modal: modalSlice.reducer,
        progressIndicator: progressIndicatorSlice.reducer,
        renterPreference: renterPreferenceSlice.reducer,
        requestForQuote: requestForQuoteSlice.reducer,
        userStatus: userStatusSlice.reducer,
        addShip: addShipSlice.reducer,
        documentName: documentNameSlice.reducer,
        termsStatus: termsStatusSlice.reducer,
        userLanguage: userLanguageSlice.reducer,
        userLocation: userLocationSlice.reducer,
        shipSpecSlice: shipSpecSlice.reducer,
        notifBadgeSlice: notifBadgeSlice.reducer,
        networkError: networkErrorSlice.reducer,
    },
});

export default store;
