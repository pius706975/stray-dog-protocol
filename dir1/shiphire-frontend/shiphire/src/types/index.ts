export * from './Components'
export * from './Navigators'
export * from './ReduxToolkit'
export * from './Screens'
export * from './Services'