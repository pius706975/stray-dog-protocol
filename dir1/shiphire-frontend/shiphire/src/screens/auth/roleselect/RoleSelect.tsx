import React from 'react';
import { useWindowDimensions } from 'react-native';
import { View } from 'react-native-ui-lib';
import { useDispatch } from 'react-redux';
import { Button, CustomText, ScreenLayout } from '../../../components';
import {
    ROLES,
    USERDATA,
    getDataFromLocalStorage,
    removeDataToLocalStorage,
    setDataToLocalStorage,
} from '../../../configs';
import { useSubmitUserRole } from '../../../hooks';
import {
    modalSlice,
    progressIndicatorSlice,
    userStatusSlice,
} from '../../../slices';
import { RoleSelectProps, SubmitUserRoleRequest } from '../../../types';
import { handleAxiosError } from '../../../utils';
import CustomCheckbox from './component/CustomCheckbox';

const RoleSelect: React.FC<RoleSelectProps> = () => {
    const dispatch = useDispatch();
    const { hideModal, showModal } = modalSlice.actions;
    const { showProgressIndicator, hideProgressIndicator } =
        progressIndicatorSlice.actions;
    const {
        setRoleSubmitted: roleIsSubmitted,
        setPreferencesNotSubmitted,
        setRoleOwner,
    } = userStatusSlice.actions;
    const { width } = useWindowDimensions();
    const mutationSubmitUserRole = useSubmitUserRole();
    const [renterChecked, setRenterChecked] = React.useState<boolean>(false);
    const [shipOwnerChecked, setShipOwnerChecked] =
        React.useState<boolean>(false);
    const [userData, setUserData] = React.useState<string>('');
    const [roleSubmitted, setRoleSubmitted] =
        React.useState<SubmitUserRoleRequest>({
            roleSubmitted: '',
        });
    const userName = userData?.split(' ')[0];

    const handleRenterCheckboxPress = () => {
        setRenterChecked(true);
        setShipOwnerChecked(false);
        setRoleSubmitted({ roleSubmitted: 'renter' });
    };

    const handleShipOwnerCheckboxPress = () => {
        setRenterChecked(false);
        setShipOwnerChecked(true);
        setRoleSubmitted({ roleSubmitted: 'shipOwner' });
    };
    const handleRoleConfirm = () => {
        if (renterChecked == false && shipOwnerChecked == false) {
            dispatch(
                showModal({
                    status: 'failed',
                    text: 'Please choose your role',
                }),
            );
            setTimeout(() => {
                dispatch(hideModal());
            }, 4000);
            return;
        }
        dispatch(setPreferencesNotSubmitted());
        dispatch(showProgressIndicator());
        mutationSubmitUserRole.mutate(roleSubmitted, {
            onSuccess: resp => {
                removeDataToLocalStorage(ROLES);
                setDataToLocalStorage(ROLES, { roles: resp.data.data.roles });
                dispatch(
                    showModal({
                        status: 'success',
                        text: `Hello new ${resp.data.data.roles}`,
                    }),
                );
                setTimeout(() => {
                    dispatch(hideModal());
                    dispatch(hideProgressIndicator());
                    resp.data.data.roles === 'renter'
                        ? dispatch(roleIsSubmitted())
                        : (dispatch(setRoleOwner()),
                          dispatch(roleIsSubmitted()));
                }, 2000);
            },
            onError: err => {
                dispatch(
                    showModal({
                        status: 'failed',
                        text: 'Failed to submit roles, please try again later',
                    }),
                );
                setTimeout(() => {
                    dispatch(hideProgressIndicator());
                    dispatch(hideModal());
                }, 2000);
                handleAxiosError(err);
            },
        });
    };

    React.useEffect(() => {
        getDataFromLocalStorage(USERDATA).then(resp => {
            if (resp) {
                setUserData('name' in resp ? resp.name : '');
            }
        });
    }, []);

    return (
        <ScreenLayout
            flex
            center
            spread
            backgroundColor="light"
            paddingV={113}
            testId="RoleSelectScreen">
            <View center>
                <CustomText
                    fontFamily="bold"
                    fontSize="xxl"
                    color="primaryColor">
                    Helo {userName},
                </CustomText>
                <CustomText
                    fontFamily="bold"
                    fontSize="xxl"
                    color="primaryColor">
                    Welcome to ShipHire!
                </CustomText>
                <View marginT-20 style={{ marginBottom: 100 }}>
                    <CustomText
                        fontFamily="regular"
                        fontSize="lg"
                        color="darkTextColor">
                        Please choose your role
                    </CustomText>
                </View>
                <View row>
                    <CustomCheckbox
                        checked={renterChecked}
                        onPress={handleRenterCheckboxPress}
                        title="Renter"
                        testId="renterCB"
                    />
                    <CustomCheckbox
                        checked={shipOwnerChecked}
                        onPress={handleShipOwnerCheckboxPress}
                        title="Ship Owner"
                        testId="shipOwnerCB"
                    />
                </View>
            </View>
            <View
                style={{
                    width: width / 1.2,
                }}>
                <Button title="Confirm" onSubmit={handleRoleConfirm} />
            </View>
        </ScreenLayout>
    );
};

export default RoleSelect;
