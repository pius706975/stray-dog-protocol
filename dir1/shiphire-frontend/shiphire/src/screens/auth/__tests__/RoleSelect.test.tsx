import AsyncStorage from "@react-native-async-storage/async-storage";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { fireEvent, render, screen, waitFor } from "@testing-library/react-native";
import MockAdapter from "axios-mock-adapter";
import { QueryClient, QueryClientProvider } from "react-query";
import { Provider } from "react-redux";
import { ROLES, TOKEN, USERDATA } from "../../../configs";
import httpRequest from "../../../services/api";
import store from "../../../store";
import { RootParamList } from "../../../types";
import RoleSelect from "../roleselect/RoleSelect";
import CustomCheckbox from "../roleselect/component/CustomCheckbox";

const queryClient = new QueryClient();
const RootStack = createNativeStackNavigator<RootParamList>()
const mockAdapter = new MockAdapter(httpRequest);

const rootStackMockComponent = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <RootStack.Navigator>
                    <RootStack.Screen name="RoleSelect" component={RoleSelect} />
                </RootStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
)

const mockRoleSelectRespSuccess = {
    status: "success",
    data: {
        _id: "mockId",
        name: "mockName",
        email: "mockEmail",
        phoneNumber: "mockPhoneNumber",
        firebaseId: "mockFirebaseId",
        isVerified: false,
        roles: "renter",
    }
}

const mockRoleSelectRespfail = {
    status: 'fail',
    message: 'User not found'
}

beforeEach(() => {
    AsyncStorage.setItem(USERDATA, JSON.stringify({
        name: "Usmanul",
        email: "usman@email.com",
        password: "test",
        isVerified: false,
        isCompanySubmitted: false,
    }))
    AsyncStorage.setItem(TOKEN, JSON.stringify({
        accessToken: "test",
        refreshToken: "test"
    }))
    mockAdapter.reset();
});

describe("Testing Role Select screen", () => {
    describe("Snapshot Testing", () => {
        it('should render Role Select screen correctly', async () => {
            const roleSelectScreen = render(rootStackMockComponent).toJSON()

            expect(roleSelectScreen).toMatchSnapshot()
        })
        describe("Component on RoleSelect screen", () => {
            describe("CustomCheckBox component", () => {
                it('should render correctly', () => {
                    const customCheckBoxComponent = render(
                        <CustomCheckbox
                            checked
                            onPress={() => { }}
                            title=""
                            testId=""
                        />
                    ).toJSON()

                    expect(customCheckBoxComponent).toMatchSnapshot()
                })
            })
        })
    })

    describe("Unit test", () => {
        it("should render 2 Custom checkbox component with title Renter and Ship Owner at unchecked condition", async () => {
            render(rootStackMockComponent)

            const renterCheckBox = await screen.findByTestId("renterCB")
            const shipOwnerCheckBox = await screen.findByTestId("shipOwnerCB")

            expect(renterCheckBox.props.accessibilityState.checked).toBe(false)
            expect(shipOwnerCheckBox.props.accessibilityState.checked).toBe(false)
        })

        it("should checked renter checkbox and unchecked ship owner checkbox when renter checkbox pressed", async () => {
            render(rootStackMockComponent)

            const renterCheckBox = await screen.findByTestId("renterCB")
            const shipOwnerCheckBox = await screen.findByTestId("shipOwnerCB")

            expect(renterCheckBox.props.accessibilityState.checked).toBe(false)
            expect(shipOwnerCheckBox.props.accessibilityState.checked).toBe(false)

            fireEvent.press(renterCheckBox)
            expect(renterCheckBox.props.accessibilityState.checked).toBe(true)
            expect(shipOwnerCheckBox.props.accessibilityState.checked).toBe(false)
        })

        it("should checked ship owner checkbox and unchecked renter checkbox when ship owner checkbox pressed", async () => {
            render(rootStackMockComponent)

            const renterCheckBox = await screen.findByTestId("renterCB")
            const shipOwnerCheckBox = await screen.findByTestId("shipOwnerCB")

            expect(renterCheckBox.props.accessibilityState.checked).toBe(false)
            expect(shipOwnerCheckBox.props.accessibilityState.checked).toBe(false)

            fireEvent.press(shipOwnerCheckBox)
            expect(renterCheckBox.props.accessibilityState.checked).toBe(false)
            expect(shipOwnerCheckBox.props.accessibilityState.checked).toBe(true)
        })

        it("should show error modal when trying to submit role but not checked any checkbox", async () => {
            jest.useFakeTimers()
            render(rootStackMockComponent)

            const renterCheckBox = await screen.findByTestId("renterCB")
            const shipOwnerCheckBox = await screen.findByTestId("shipOwnerCB")
            const submitBtn = await screen.findByText("Confirm")

            expect(renterCheckBox.props.accessibilityState.checked).toBe(false)
            expect(shipOwnerCheckBox.props.accessibilityState.checked).toBe(false)

            fireEvent.press(submitBtn)

            await waitFor(() => {
                const modalState = store.getState().modal

                expect(modalState).toEqual({
                    visible: true,
                    status: 'failed',
                    text: 'Please choose your role'
                })
            })

            await waitFor(() => {
                const modalState = store.getState().modal

                jest.advanceTimersByTime(4000)
                expect(modalState).toEqual({
                    visible: false,
                    status: undefined,
                    text: ''
                })
            })

            jest.useRealTimers()
        })

        it('should show success modal when trying to submit role and checked renter checkbox', async () => {
            mockAdapter.onPost('/user/submit-user-role').reply(200, mockRoleSelectRespSuccess)

            jest.useFakeTimers()

            render(rootStackMockComponent)

            const renterCheckBox = await screen.findByTestId("renterCB")
            const submitBtn = await screen.findByText("Confirm")

            fireEvent.press(renterCheckBox)
            fireEvent.press(submitBtn)

            await waitFor(() => {
                const modalState = store.getState().modal
                const isPreferencesSubmittedState = store.getState().userStatus.isPreferencesSubmitted
                const showProgressIndicatorState = store.getState().progressIndicator.visible

                expect(modalState).toEqual({
                    visible: true,
                    status: 'success',
                    text: `Hello new ${mockRoleSelectRespSuccess.data.roles}`
                })

                expect(isPreferencesSubmittedState).toBe(false)
                expect(showProgressIndicatorState).toBe(true)
                expect(AsyncStorage.removeItem).toBeCalledWith(ROLES)
                expect(AsyncStorage.setItem).toBeCalledWith(ROLES, JSON.stringify({ roles: mockRoleSelectRespSuccess.data.roles }))
            })

            await waitFor(() => {
                const isRoleSubmittedState = store.getState().userStatus.isRoleSubmitted
                const showProgressIndicatorState = store.getState().progressIndicator.visible

                jest.advanceTimersByTime(2000)
                expect(isRoleSubmittedState).toBe(true)
                expect(showProgressIndicatorState).toBe(false)
            })

            jest.useRealTimers()
        })

        it('should handle axios error when some error happend while submitting role', async () => {
            mockAdapter.onPost('/user/submit-user-role').reply(404, mockRoleSelectRespfail)

            render(rootStackMockComponent)

            const renterCheckBox = await screen.findByTestId("renterCB")
            const submitBtn = await screen.findByText("Confirm")

            fireEvent.press(renterCheckBox)
            fireEvent.press(submitBtn)

            await waitFor(() => {
                const handleAxiosError = jest.spyOn(console, 'log')

                expect(handleAxiosError).toBeCalled()
            })
        })
    })
})