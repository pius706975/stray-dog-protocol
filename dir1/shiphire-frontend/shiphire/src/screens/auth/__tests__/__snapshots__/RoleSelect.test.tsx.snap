// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Role Select screen Snapshot Testing Component on RoleSelect screen CustomCheckBox component should render correctly 1`] = `
<View
  accessibilityRole="checkbox"
  accessibilityState={
    {
      "busy": undefined,
      "checked": true,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  checkedIcon={
    <Text
      style={
        {
          "color": "#f0f6fb",
          "fontFamily": "Kanit-Medium",
          "fontSize": 20,
        }
      }
    >
      
    </Text>
  }
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignContent": "flex-start",
      "backgroundColor": "#216dab",
      "borderColor": "#ffffff",
      "borderRadius": 10,
      "flexDirection": "row",
      "height": 76,
      "justifyContent": "center",
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 10,
      "width": 153,
    }
  }
  testID=""
  uncheckedIcon={
    <Text
      style={
        {
          "color": "#216dab",
          "fontFamily": "Kanit-Medium",
          "fontSize": 20,
        }
      }
    >
      
    </Text>
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        [
          {
            "backgroundColor": "transparent",
            "color": "#20303C",
            "writingDirection": "ltr",
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "color": "#f0f6fb",
            "fontFamily": "Kanit-Medium",
            "fontSize": 20,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`Testing Role Select screen Snapshot Testing should render Role Select screen correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="RoleSelectScreen"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "flexGrow": 1,
                  "gap": undefined,
                  "justifyContent": "space-between",
                  "marginBottom": undefined,
                  "padding": undefined,
                  "paddingVertical": 113,
                }
              }
            >
              <View>
                <View
                  center={true}
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {},
                      {},
                      {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={0}
                    style={
                      {
                        "borderBottomWidth": 0,
                        "borderColor": undefined,
                        "color": "#216dab",
                        "fontFamily": "Kanit-Bold",
                        "fontSize": 30,
                        "lineHeight": undefined,
                        "textAlign": undefined,
                      }
                    }
                  >
                    Helo 
                    ,
                  </Text>
                  <Text
                    numberOfLines={0}
                    style={
                      {
                        "borderBottomWidth": 0,
                        "borderColor": undefined,
                        "color": "#216dab",
                        "fontFamily": "Kanit-Bold",
                        "fontSize": 30,
                        "lineHeight": undefined,
                        "textAlign": undefined,
                      }
                    }
                  >
                    Welcome to ShipHire!
                  </Text>
                  <View
                    marginT-20={true}
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {},
                        {
                          "marginTop": 20,
                        },
                        {},
                        {
                          "marginBottom": 100,
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={0}
                      style={
                        {
                          "borderBottomWidth": 0,
                          "borderColor": undefined,
                          "color": "#4d5963",
                          "fontFamily": "Kanit-Regular",
                          "fontSize": 20,
                          "lineHeight": undefined,
                          "textAlign": undefined,
                        }
                      }
                    >
                      Please choose your role
                    </Text>
                  </View>
                  <View
                    row={true}
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {},
                        {},
                        {
                          "flexDirection": "row",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      accessibilityRole="checkbox"
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": false,
                          "disabled": false,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      checkedIcon={
                        <Text
                          style={
                            {
                              "color": "#f0f6fb",
                              "fontFamily": "Kanit-Medium",
                              "fontSize": 20,
                            }
                          }
                        >
                          Renter
                        </Text>
                      }
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignContent": "flex-start",
                          "backgroundColor": "#d7edff",
                          "borderColor": "#216dab",
                          "borderRadius": 10,
                          "borderWidth": 2,
                          "flexDirection": "row",
                          "height": 76,
                          "justifyContent": "center",
                          "margin": 5,
                          "marginLeft": 10,
                          "marginRight": 10,
                          "padding": 10,
                          "width": 153,
                        }
                      }
                      testID="renterCB"
                      uncheckedIcon={
                        <Text
                          style={
                            {
                              "color": "#216dab",
                              "fontFamily": "Kanit-Medium",
                              "fontSize": 20,
                            }
                          }
                        >
                          Renter
                        </Text>
                      }
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "backgroundColor": "transparent",
                                "color": "#20303C",
                                "writingDirection": "ltr",
                              },
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              {
                                "color": "#216dab",
                                "fontFamily": "Kanit-Medium",
                                "fontSize": 20,
                              },
                            ]
                          }
                        >
                          Renter
                        </Text>
                      </View>
                    </View>
                    <View
                      accessibilityRole="checkbox"
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": false,
                          "disabled": false,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      checkedIcon={
                        <Text
                          style={
                            {
                              "color": "#f0f6fb",
                              "fontFamily": "Kanit-Medium",
                              "fontSize": 20,
                            }
                          }
                        >
                          Ship Owner
                        </Text>
                      }
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignContent": "flex-start",
                          "backgroundColor": "#d7edff",
                          "borderColor": "#216dab",
                          "borderRadius": 10,
                          "borderWidth": 2,
                          "flexDirection": "row",
                          "height": 76,
                          "justifyContent": "center",
                          "margin": 5,
                          "marginLeft": 10,
                          "marginRight": 10,
                          "padding": 10,
                          "width": 153,
                        }
                      }
                      testID="shipOwnerCB"
                      uncheckedIcon={
                        <Text
                          style={
                            {
                              "color": "#216dab",
                              "fontFamily": "Kanit-Medium",
                              "fontSize": 20,
                            }
                          }
                        >
                          Ship Owner
                        </Text>
                      }
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "backgroundColor": "transparent",
                                "color": "#20303C",
                                "writingDirection": "ltr",
                              },
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              {
                                "color": "#216dab",
                                "fontFamily": "Kanit-Medium",
                                "fontSize": 20,
                              },
                            ]
                          }
                        >
                          Ship Owner
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "width": 625,
                      },
                    ]
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#216dab",
                        "borderRadius": 6,
                        "elevation": 2,
                        "flexDirection": "row",
                        "gap": 0,
                        "justifyContent": "center",
                        "paddingHorizontal": 10,
                        "paddingVertical": 6,
                      }
                    }
                  >
                    <Text
                      numberOfLines={0}
                      style={
                        {
                          "borderBottomWidth": 0,
                          "borderColor": undefined,
                          "color": "#f0f6fb",
                          "fontFamily": "Kanit-Medium",
                          "fontSize": 18,
                          "lineHeight": undefined,
                          "textAlign": undefined,
                        }
                      }
                    >
                      Confirm
                    </Text>
                    <View
                      style={
                        [
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "position": "absolute",
                            "right": 20,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="RoleSelect"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
